main.go

```diff
@@ -185,3 +224,77 @@ func setupWebhooks(mgr ctrl.Manager) {
 func concurrency(c int) controller.Options {
 	return controller.Options{MaxConcurrentReconciles: c}
 }
+
+// GetTLSOptionOverrideFuncs returns a list of TLS configuration overrides to be used
+// by the webhook server.
+func GetTLSOptionOverrideFuncs(options TLSOptions) ([]func(*tls.Config), error) {
+	var tlsOptions []func(config *tls.Config)
+	var TLSMinVersionIndex, TLSMaxVersionIndex int
+	for tlsVersionIndex, TLSVersion := range cliflag.TLSPossibleVersions() {
+		if options.TLSMinVersion == TLSVersion {
+			TLSMinVersionIndex = tlsVersionIndex
+		}
+		if options.TLSMaxVersion == TLSVersion {
+			TLSMaxVersionIndex = tlsVersionIndex
+		}
+	}
+	// Check if tls min version flag value is greater than max version
+	if TLSMinVersionIndex > TLSMaxVersionIndex &&
+		TLSMaxVersionIndex != -1 &&
+		TLSMinVersionIndex != -1 {
+		return nil, fmt.Errorf("TLS version flag min version is greater than max version")
+	}
+
+	tlsMinVersion, err := cliflag.TLSVersion(options.TLSMinVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MinVersion = tlsMinVersion
+	})
+
+	tlsMaxVersion, err := cliflag.TLSVersion(options.TLSMaxVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MaxVersion = tlsMaxVersion
+	})
+	// Cipher suites should not be set if empty.
+	if options.TLSMinVersion == "VersionTLS13" && options.TLSMaxVersion == "VersionTLS13" {
+		log.Print("cipher suite flag set to null for TLS version 1.3")
+		options.TLSCipherSuites = ""
+	}
+
+	if options.TLSCipherSuites != "" {
+		tlsCipherSuites := strings.Split(options.TLSCipherSuites, ",")
+		suites, err := cliflag.TLSCipherSuites(tlsCipherSuites)
+		if err != nil {
+			return nil, err
+		}
+
+		insecureCipherValues := cliflag.InsecureTLSCipherNames()
+		for _, cipher := range tlsCipherSuites {
+			for _, insecureCipherName := range insecureCipherValues {
+				if insecureCipherName == cipher {
+					log.Printf("use of insecure cipher '%s' detected.", cipher)
```

Comment:
```suggestion
					setupLog.Info("warning: use of insecure cipher '%s' detected.", cipher)
```

Reply:
@lentzi90  do you have any idea how can i fetch these logs inside a buffer in my test files so that I can test them. I was able to do that with the logger library but trying the same here doesn't seem to capture the logs.

Reply:
Oh, you are checking the log output. I'm not sure if that is possible in this way. Maybe this could be used?
https://pkg.go.dev/k8s.io/klog/v2/ktesting

Anyway, I think it may be better to not check the log output and just see that the function returns the expected configuration

Reply:
I was able to fix this logging issue without using ktesting since this library is experimental. I was calling the logger using an incorrect way in the tests. Actually there is a test which detect if insecure cipher are passed so for that checking the output doesn't seem the right way since the output will contain those values. Other than that the output of this function under test  []func(*tls.Config), error is an array of functions with memory pointers which can't be tested. Returning the configuration from this function in struct form would be redundant since It won't be used anywhere in the original code. Going through different articles such as 
https://goutamtadi.medium.com/unit-testing-logs-with-logr-logger-in-golang-59cfa07f9178
suggested this way to do this as well.

Reply:
Nice! This looks like a good solution to me!